src/parser/Parser.ts

```diff
@@ -24,797 +26,872 @@ type BlockTerminator =
     Lexeme.EndSub |
     Lexeme.EndFunction;
 
-let current: number;
-let tokens: ReadonlyArray<Token>;
+/** The results of a Parser's parsing pass. */
+interface ParseResults {
+    /** The statements produced by the parser. */
+    statements: ReadonlyArray<Stmt.Statement>,
```

Comment:
It appears that you're actually returning a regular array in the parse method. Making this a `ReadonlyArray` causes me to have to use the same `ReadonlyArray` type in my code, which doesn't seem to provide much benefit. I think it's probably safe enough to just make this an array, and it's my own dumb fault if I screw up my `statements` array.

Reply:
To be honest ReadonlyArray was more about keeping me from screwing things up within the runtime. I’ll look into making the lexer, parser, and friends accept readonly arrays (to keep me from mutating within them)?but return plain-old arrays.

Thanks for the feedback!